pymc_marketing/customer_choice/nested_logit.py

```diff
@@ -540,37 +540,52 @@ def make_exp_nest(
         else:
             parent = None
         y_nest = U[:, nest_indices[level][nest]]
+        n_alts_in_nest = len(nest_indices[level][nest])
+
         if W is None:
-            w_nest = pm.math.zeros((N, len(self.alternatives)))
+            w_nest = pm.math.zeros((N, n_alts_in_nest))
         else:
-            betas_fixed_temp = betas_fixed[nest_indices[level][nest], :]
-            betas_fixed_temp = pt.set_subtensor(betas_fixed_temp[-1], 0)
+            # Select rows for this nest's alternatives
+            nest_alt_indices = nest_indices[level][nest]
+            if n_alts_in_nest == 1:
+                # For single alternative nests, explicitly maintain 2D shape
```

Comment:
Use expand dims instead?

Reply:
mmm I still do not know either, this is me trying to make broadcasting work again 🙈